pkg/apis/azuredisk/v1alpha1/types.go

```diff
@@ -23,23 +23,44 @@ import (
 // +genclient
 // +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
 
-// AzVolume is a specification for a AzVolume resource
+// AzVolume is a specification for an AzVolume resource
 type AzVolume struct {
 	metav1.TypeMeta   `json:",inline"`
 	metav1.ObjectMeta `json:"metadata,omitempty"`
 
-	Spec   AzVolumeSpec   `json:"spec"`
-	Status AzVolumeStatus `json:"status"`
+	// spec defines the desired state of an AzVolume.
+	// Required.
+	Spec AzVolumeSpec `json:"spec"`
+	// status represents the current state of AzVolume.
+	// Nil status indicates that the underlying volume has not yet been provisioned
+	// +optional
+	Status *AzVolumeStatus `json:"status,omitempty"`
 }
 
-// AzVolumeSpec is the spec for a AzVolume resource
+// AzVolumeSpec is the spec for an AzVolume resource
 type AzVolumeSpec struct {
 	UnderlyingVolume string `json:"underlyingVolume"`
+
+	MaxMountReplicaCount int `json:"maxMountReplicaCount"`
+	//The suggested name for the storage space
+	Name string `json:"name"`
+	//The capabilities that the volume MUST have
+	VolumeCapability *VolumeCapability `json:"volumeCapability"`
+	//The capacity of the storage
+	CapacityRange *CapacityRange `json:"capacityRange"`
+	//Parameters for the volume
+	//+optional
+	Parameters map[string]string `json:"parameters,omitempty"`
 }
 
-// AzVolumeStatus is the status for a AzVolume resource
+// AzVolumeStatus is the status for an AzVolume resource
 type AzVolumeStatus struct {
+	//Current state of the AzVolume
+	//+optional
 	State string `json:"state,omitempty"`
+	//Current volume ID for the underlying volume
+	//+optional
+	UnderlyingVolume string `json:"underlyingVolume"`
```

Comment:
Do we have scenarios where the the value under `UnderlyingVolume` is changed? If not, we might be able to get rid of this field in `Status`.

Reply:
I see what you're saying. The value of underlyingVolume will eventually be set by the CreateVolume function with the Volume ID/diskURI thats returned from creating, so it needs to be updated at some point. My thinking was that it would be easiest to set it as part of status, but I could also see instead about updating the Spec field with that value

Reply:
I think I may have led you astray here. Looking at the current `CreationVolume` implementation, the name of the PVC is known prior to creation. The resource URI (volume ID) can be constructed at `AzVolume` creation time, and so should just be part of the `Spec`.

I wonder if we should call this field `VolumeID` to be consistent with the CSI terminology or `DiskURI` to be clear about what it really is. Personally, I find `UnderlyingVolume` to be a bit nebulous.

Reply:
Sounds good, I've gone ahead and changed the name to VolumeID to be more clear, and included it in only the Spec